cpu/sources/layout7/memory_access_utils.move

```diff
@@ -0,0 +1,29 @@
+module cpu_addr::memory_access_utils_7 {
+    use std::vector::{borrow, push_back};
+
+    // This line is used for generating constants DO NOT REMOVE!
+    // 9
+    const ENOT_ENOUGH_FRI_STEPS: u64 = 0x9;
+    // 8
+    const ETOO_MANY_FRI_STEPS: u64 = 0x8;
+    // 10
+    const MAX_FRI_STEPS: u64 = 0xa;
+    // 5
+    const PROOF_PARAMS_FRI_STEPS_OFFSET: u64 = 0x5;
+    // 4
+    const PROOF_PARAMS_N_FRI_STEPS_OFFSET: u64 = 0x4;
+    // End of generating constants!
+
+    public fun get_fri_step_sizes(proof_params: &vector<u256>): vector<u256> {
+        let n_fri_steps = (*borrow(proof_params, PROOF_PARAMS_N_FRI_STEPS_OFFSET) as u64);
+        assert!(n_fri_steps <= MAX_FRI_STEPS, ETOO_MANY_FRI_STEPS);
+        assert!(n_fri_steps > 1, ENOT_ENOUGH_FRI_STEPS);
+
+        let fri_step_sizes = vector[];
+        for (i in 0..n_fri_steps) {
+            push_back(&mut fri_step_sizes, *borrow(proof_params, PROOF_PARAMS_FRI_STEPS_OFFSET + i));
+        };
+
+        fri_step_sizes
+    }
```

Comment:
public fun get_fri_step_sizes(proof_params: &vector<u256>): vector<u256> {
        let n_fri_steps = (*borrow(proof_params, PROOF_PARAMS_N_FRI_STEPS_OFFSET) as u64);
        assert!(n_fri_steps <= MAX_FRI_STEPS, ETOO_MANY_FRI_STEPS);
        assert!(n_fri_steps > 1, ENOT_ENOUGH_FRI_STEPS);
        assert!(PROOF_PARAMS_FRI_STEPS_OFFSET + n_fri_steps == length(proof_params), EPROOF_PARAMS_INVALID_SIZE);

        let fri_step_sizes = *proof_params;
        trim_head(&mut fri_step_sizes, PROOF_PARAMS_FRI_STEPS_OFFSET);
        fri_step_sizes
    }

Reply:
In terms of gas const, it will be better